examples/k3s/composition.yaml

```diff
@@ -0,0 +1,149 @@
+apiVersion: apiextensions.crossplane.io/v1
+kind: Composition
+metadata:
+  name: k3sclusters.deployments.equinix.com
+spec:
+  writeConnectionSecretsToNamespace: crossplane-system
+  compositeTypeRef:
+    apiVersion: deployments.equinix.com/v1alpha1
+    kind: K3sCluster
+  resources:
+    - name: k3s-project
+      base:
+        apiVersion: metal.equinix.jet.crossplane.io/v1alpha1
+        kind: Project
+      patches:
+        - type: FromCompositeFieldPath
+          fromFieldPath: metadata.name
+          toFieldPath: spec.forProvider.name
+    - name: k3s-userdata
+      base:
+        apiVersion: v1
+        kind: Secret
+        type: Opaque
+        metadata:
+          namespace: crossplane-system
+      patches:
+        - type: CombineFromComposite
+          toFieldPath: metadata.name
+          combine:
+            variables:
+              - fromFieldPath: metadata.name
+            strategy: string
+            string:
+              fmt: "k3s-userdata-%s"
+        - type: FromCompositeFieldPath
+          toFieldPath: metadata.namespace
+          fromFieldPath: spec.secretNamespace
+        - type: CombineFromComposite
+          toFieldPath: data.userdata
+          combine:
+            variables:
+              - fromFieldPath: spec.k3stoken
+            strategy: string
+            string:
+              fmt: |
+                #cloud-config
+                write_files:
+                #- permissions: '0660'
+                #  path: /etc/rancher/k3s/config.yaml
+                #  content: |
+                #    # k3s config  
+                runcmd:
+                - sudo apt update
+                - sudo apt upgrade -y
+                - curl -sfL https://get.k3s.io | \
```

Comment:
Certificates for the Kubernetes cluster should be sourced from Secrets which could then be managed by cert-manager.  This will keep the user in control of secrets rather than having them only available within the servers (and therefor difficult to get access to from a Crossplane composition).

Crossplane compositions would also benefit from this K8s managed certificate for managing helm installs within the created cluster.